### PR TITLE
clang check modernize-use-override

### PR DESCRIPTION
**Pull request by davidlange6**
also for comment - this is what clang-tidy check modernize-use-override does

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by slava77** [comment]
-1
just to have it off the reco list
the diff is too large to inspect reco files.

From the first few files that are displayed on the web interface, the change seems OK, although I'd probably keep the "virtual" specifier if it's already there.
Even though it's redundant, it was our code rule 3.26.

File: Alignment/APEEstimation/plugins/ApeEstimator.cc [hunk 1]
```diff
@@ -129,7 +129,7 @@
 class ApeEstimator : public edm::EDAnalyzer {
```
Comment (ianna) [inline Alignment/APEEstimation/plugins/ApeEstimator.cc]: @davidlange6 - this should be thread-safe analyzer. The rest of API changes would follow.

File: Alignment/APEEstimation/plugins/ApeEstimatorSummary.cc [hunk 2]
```diff
@@ -62,13 +62,13 @@
 class ApeEstimatorSummary : public edm::EDAnalyzer {
```
Comment (ianna) [inline Alignment/APEEstimation/plugins/ApeEstimatorSummary.cc]: @davidlange6 - the same here - it should be a thread-safe analyzer.

File: Alignment/CocoaApplication/interface/CocoaAnalyzer.h [hunk 3]
```diff
@@ -44,10 +44,10 @@ class CocoaAnalyzer : public edm::one::EDAnalyzer<edm::one::SharedResources>
   
   explicit  CocoaAnalyzer(edm::ParameterSet const& p);
   explicit  CocoaAnalyzer(int i) { }
-  virtual ~ CocoaAnalyzer() { }
+  ~ CocoaAnalyzer() override { }
```
Comment (ianna) [inline Alignment/CocoaApplication/interface/CocoaAnalyzer.h]: @davidlange6 - just a general question: is COCOA still used/alive?

File: Alignment/CommonAlignmentMonitor/plugins/AlignmentMonitorAsAnalyzer.cc [hunk 4]
```diff
@@ -66,15 +66,15 @@
 class AlignmentMonitorAsAnalyzer : public edm::EDAnalyzer {
```
Comment (ianna) [inline Alignment/CommonAlignmentMonitor/plugins/AlignmentMonitorAsAnalyzer.cc]: it should be a thread-safe analyzer

**Review by ianna — CHANGES_REQUESTED** [review]
I did not manage to get to the Geometry category - there are way to many files.

**Comment by davidlange6** [comment]
this PR has only one purpose - to show what modernize-use-override does to our code (and it won't be merged in this form). Any effect to fix other old constructs existing in CMSSW is a separate discussion and PR......

but right, I think to see everything you need to just go look at my branch or checkout locally. Maybe some of the other tools will have smaller diffs..

**Comment by ianna** [comment]
@davidlange6 - IMHO it's better to apply changes to smaller chunks of code. I'm afraid if I check the branch out, it will take a while to compile it.

**Comment by davidlange6** [comment]
I don't think I have that technology (things will be missed). eventually we will do things in steps.

**Comment by ianna** [comment]
-1

Please, use https://github.com/cms-sw/cmssw/pull/19610 to remove the need for geometry signature

**Pull request closed without merging**